ts/Dashboards/Components/KPIComponent/KPIComponent.ts

```diff
@@ -405,6 +406,118 @@ class KPIComponent extends Component {
+     * Calculates value average based on the provided values.
+     *
+     * @param values
+     * All particular column values.
+     *
+     * @returns
+     * The calculated average value.
+     *
+     * @internal
+     */
+    private calculateAverageValue(values: number[]): number {
+        const valueSum = this.calculateSumValue(values);
+        return valueSum / values.length;
+    }
+
+    /**
+     * Calculates value median based on the provided values.
+     *
+     * @param values
+     * All particular column values.
+     *
+     * @returns
+     * The calculated median value.
+     *
+     * @internal
+     */
+    private calculateMedianValue(values: number[]): number {
+        const sortedValues = Array.from(values).sort((a, b): number => a - b);
+        const middleValue = Math.floor(sortedValues.length / 2);
+        const middleSortedValue = sortedValues[middleValue];
+
+        return sortedValues.length % 2 === 0 ?
+            (sortedValues[middleValue - 1] + middleSortedValue) / 2 :
+            middleSortedValue;
+    }
+
+    /**
+     * Gets the proper calculation value, according to the provided
+     * calculateValueAs option.
+     *
+     * @returns
+     * The calculated value or 0 if can't perform the calculation. Can be a 
+     * number internally, or a string from the callback function.
+     *
+     * @internal
+     */
+    private getCalculatedValue(): string|number {
+        const calculateValueAs = this.options.calculateValueAs;
+
+        const connector = this.getFirstConnector();
+        const table = connector?.table.modified;
+        const column = table?.getColumn(this.options.columnName);
+        
+        // Parse into numerical values to perform calculations.
+        const parsedColumnValues =
+            column?.map((value): number => Number(value));
+
+        let calculatedValue: string|number = 0;
+
+        // Provide external calculations without a strict type checking.
+        if (defined(parsedColumnValues) && isFunction(calculateValueAs)) {
+            calculatedValue = calculateValueAs.call(this, parsedColumnValues);
+
+        // Make sure all values are numbers to perform accurate calculations.
+        } else if (parsedColumnValues?.every(isNumber)) {
+            if (calculateValueAs === 'sum') {
+                calculatedValue = this.calculateSumValue(parsedColumnValues);
+
+            } else if (calculateValueAs === 'average') {
+                calculatedValue =
+                    this.calculateAverageValue(parsedColumnValues);
+
+            } else if (calculateValueAs === 'median') {
+                calculatedValue = this.calculateMedianValue(parsedColumnValues);
+
+            } else {
+                console.warn('Invalid calculate option value provided.'); // eslint-disable-line no-console
+            }
+        } else {
+            console.warn('All column values should be numerical to calculate.'); // eslint-disable-line no-console
+        }
+
```

Comment:
Could you refactor this if-tree a bit? I don't really like the fact that functions are defined as methods of this class. It could be a static object containing definitions of these calculations, for a given `key` of type `typeof calculateValueAs`.

Reply:
I actually considered a map but ended up with these if statements 😄. I have added changes, feel free to share your thoughts.